tools/toolchains/__init__.py

```diff
@@ -344,7 +344,7 @@ def get_symbols(self):
 
             # Config support
             if self.has_config:
-                self.symbols.append('HAVE_MBED_CONFIG_H')
+                self.symbols.append('HAVE_MBED_CONFIG')
```

Comment:
This will work, but looks confusing. You're still adding the `HAVE_MBED_CONFIG` symbol,, even though the line that sets `self.has_config` to `True` was removed (below), so this code will never be reched.